src/components/game-setup/GameStepper.js

```diff
@@ -0,0 +1,137 @@
+import * as React from 'react';
+import Box from '@mui/material/Box';
+import Stepper from '@mui/material/Stepper';
+import Step from '@mui/material/Step';
+import StepButton from '@mui/material/StepButton';
+import Button from '@mui/material/Button';
+import Typography from '@mui/material/Typography';
+import Customize from './stepper-components/Customize';
+import GameType from './stepper-components/GameType';
+import Lobby from './stepper-components/Lobby';
+
+const steps = ['Choose Game Type', 'Add Friends', 'Customize and Play'];
+
+export default function HorizontalNonLinearStepper() {
+  const [activeStep, setActiveStep] = React.useState(0);
+  const [completed, setCompleted] = React.useState({});
+
+  const totalSteps = () => {
+    return steps.length;
+  };
+
+  const completedSteps = () => {
+    return Object.keys(completed).length;
+  };
+
+  const isLastStep = () => {
+    return activeStep === totalSteps() - 1;
+  };
+
+  const allStepsCompleted = () => {
+    // Add route to game start
+    return completedSteps() === totalSteps();
+  };
+
+  const handleNext = () => {
+    const newActiveStep =
+      isLastStep() && !allStepsCompleted()
+        ? // It's the last step, but not all steps have been completed,
+          // find the first step that has been completed
+          steps.findIndex((step, i) => !(i in completed))
+        : activeStep + 1;
+    setActiveStep(newActiveStep);
+  };
+
+  const handleBack = () => {
+    setActiveStep((prevActiveStep) => prevActiveStep - 1);
+  };
+
+  /*
+  const handleCurStep = () => {
+    let curStep = activeStep;
+    return stepComponents.at(curStep);
+  };
+*/
+  const handleStep = (step) => () => {
+    if (completed.includes(step)) {
+      setActiveStep(step);
+    }
+    // Inform user they must complete step to continue - toast
+  };
+
+  const handleComplete = () => {
+    const newCompleted = completed;
+    newCompleted[activeStep] = true;
+    setCompleted(newCompleted);
+    handleNext();
+  };
+
+  // add catch here to double check reset
+  const handleReset = () => {
+    setActiveStep(0);
+    setCompleted({});
+  };
+
+  return (
+    <Box sx={{ width: '100%' }}>
+      <Stepper nonLinear activeStep={activeStep}>
+        {steps.map((label, index) => (
+          <Step key={label} completed={completed[index]}>
+            <StepButton color="inherit" onClick={handleStep(index)}>
+              {label}
+            </StepButton>
+          </Step>
+        ))}
+      </Stepper>
+      <Box
+        sx={{
+          display: 'flexbox',
+          alignCotents: 'center',
+          alignContent: 'space-between',
+          minHeight: '600px',
+          paddingTop: '100px'
+        }}>
+        {activeStep === 0 ? <GameType /> : activeStep === 1 ? <Lobby /> : <Customize />}
+      </Box>
+      <div>
+        {allStepsCompleted() ? (
+          <React.Fragment>
```

Comment:
Not a huge deal, but `<React.Fragment></React.Fragment>` is equivalent to `<></>` if you want to make it easier on yourself.